packages/lesswrong/server/resolvers/tagResolvers.ts

```diff
@@ -340,44 +502,50 @@ addGraphQLResolvers({
       if(moment.duration(moment(before).diff(after)).as('hours') > 30)
         throw new Error("TagUpdatesInTimeBlock limited to a one-day interval");
       
-      // Get revisions to tags in the given time interval
-      const revisions = await Revisions.find({
-        $and: [{
-          $or: [
-            { collectionName: "Tags", fieldName: "description" },
-            { collectionName: "MultiDocuments", fieldName: "contents" },
-          ],
-        }, {
-          $or: [
-            { "changeMetrics.added": {$gt: 0} },
-            { "changeMetrics.removed": {$gt: 0} }
-          ]
-        }],
-        editedAt: { $lt: before, $gt: after },
-      }).fetch();
-      
-      // Get root comments on tags in the given time interval
-      const rootComments = await Comments.find({
-        deleted: false,
-        postedAt: {$lt: before, $gt: after},
-        topLevelCommentId: null,
-        tagId: {$exists: true, $ne: null},
-        tagCommentType: "DISCUSSION",
-      }).fetch();
+      const rootCommentsSelector = getRootCommentsInTimeBlockSelector(before, after);
+
+      // Get
+      // - revisions to tags, lenses, and summaries in the given time interval
+      // - root comments on tags in the given time interval
+      const [revisions, rootComments] = await Promise.all([
+        context.repos.revisions.getRevisionsInTimeBlock(before, after),
+        Comments.find(rootCommentsSelector).fetch()
+      ]);
       
-      const [usersById, { lensIdsByTagId, summaryIdsByTagId }] = await Promise.all([
+      const [usersById, { lensesByTagId, summariesByTagId }] = await Promise.all([
         getRevisionAndCommentUsers(revisions, rootComments, context),
         getMultiDocumentsByTagId(revisions, context),
       ]);
 
-      const tags = await getTopLevelTags(revisions, rootComments, lensIdsByTagId, summaryIdsByTagId, context);
+      const lensIdsByTagId = mapValues(lensesByTagId, lenses => lenses.map(l => l._id!));
+      const summaryIdsByTagId = mapValues(summariesByTagId, summaries => summaries.map(s => s._id!));
+
+      const [tags, documentDeletionsByTagId] = await Promise.all([
+        getTopLevelTags(revisions, rootComments, lensIdsByTagId, summaryIdsByTagId, context),
+        getDocumentDeletionsInTimeBlock({
+          before,
+          after,
+          lensesByTagId,
+          summariesByTagId,
+          context,
+        }),
+      ]);
+
+      // We use all revisions above because we use content-less revisions to figure out
+      // what documents have been deleted (since updating the `deleted` field of a tag or multidoc creates a new revision, even if the content hasn't changed).
+      // But for actually showing revisions with diffs, we only want to show the ones that have content changes.
+      const contentfulRevisions = revisions.filter(rev => rev.changeMetrics.added > 0 || rev.changeMetrics.removed > 0);
```

Comment:
I'm wary of using `changeMetrics` this way (as opposed to string-comparing html) because I don't know for sure that there aren't any bits of content that fail to count towards change metrics. Change-metrics are trying to count changes to text content (not including html overhead); this opens the possibility of things like changing image URLs while counting as 0 characters added/removed.

Reply:
Hm, fair, though this is just mirroring the previous logic which was in the selector for fetching the relevant revisions (which was removed from the initial query, so that we could get all the content-less revisions for their documentIds, by which means we know what documentIds to look up in LWEvents).